src/main/java/fi/digitraffic/tis/vaco/crypt/EncryptionService.java

```diff
@@ -75,4 +96,81 @@ public <T> T decrypt(String cypher, Class<T> clz) {
             throw new VacoException("Failed to deserialize decrypted result", e);
         }
     }
+
+    @Value("${vaco.encryptionKeys.credentials}")
+    private String keyId;
+
+    private static KmsAsyncClient kmsAsyncClient;
+
+    public byte[] encryptBlob(AuthenticationDetails details) {
+
+            SdkBytes myBytes = SdkBytes.fromUtf8String(details.toString());
+            EncryptRequest encryptRequest = EncryptRequest.builder()
+                .keyId(keyId)
+                .plaintext(myBytes)
+                .build();
+
+            try {
+                EncryptResponse response = getAsyncClient()
+                    .encrypt(encryptRequest)
+                    .toCompletableFuture()
+                    .get();
+
+                return response.ciphertextBlob().asByteArray();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new RuntimeException("Error during encryption", e);
+            }
+    }
+
+    public byte[] decryptBlob(byte[] encryptedData) {
+
+        SdkBytes bytes = SdkBytes.fromByteArray(encryptedData);
+        logger.debug("Decrypting data: {}", Base64.getEncoder().encodeToString(encryptedData));
+        DecryptRequest decryptRequest = DecryptRequest.builder()
+            .ciphertextBlob(bytes)
+            .keyId(keyId)
+            .build();
+
+        try {
+            DecryptResponse decryptResponse = getAsyncClient()
+                .decrypt(decryptRequest)
+                .toCompletableFuture()
+                .join();
+
+            return decryptResponse.plaintext().asByteArray();
+
+        } catch (CompletionException e) {
+            throw new RuntimeException("Error occurred during decryption: " + e.getCause().getMessage(), e.getCause());
+        }
+    }
+
+    private static KmsAsyncClient getAsyncClient() {
```

Comment:
This should be created as bean in `fi.digitraffic.tis.vaco.aws.AwsConfiguration`